### DISCUSSION TITLE
Does RDKit have any function like PerceiveBondOrder?

### DISCUSSION THREAD

**Discussion by jennyfothergill** [rdkit/rdkit#4082] [Q&A]
If I give RDKit information about the atoms and the bonds between them, does RDKit have any way of inferring the bond order? Similar to http://openbabel.org/wiki/Bond_Orders? 

Thanks!

**Comment by bp-kelley** [comment]
This article from Pat Walters might help:

http://practicalcheminformatics.blogspot.com/2018/09/assigning-bond-orders-to-pdb-ligands.html?m=1

----
Brian Kelley

**Reply by jennyfothergill** [reply]
Thank you for your help. If I understand this snippet correctly, I would need the SMILES strings for the structure in order to assign the bond order. Unfortunately, I am trying to use RDKit for SMILES matching (i.e., to get the SMILES string of the structure when I only have element type, position, and bond information (except for bond order)), so this wouldn't work for me.
